astropy/visualization/time.py

```diff
@@ -0,0 +1,218 @@
+                        times.append(datetime(year=ymin + month // 12,
+                                              month=month % 12, day=1))
+
+                # Convert back to MJD
+                values = Time(times, scale=self._converter.scale).mjd
+
+                # Get rid of values outside of the input interval
+                values = values[(values >= vmin) & (values <= vmax)]
+
+                return values
+
+            elif vrange > 1:  # greater than a day
+
+                self.set_params(steps=[1, 2, 5, 10])
+                return super().tick_values(vmin, vmax)
+
+            else:
+
+                # Determine ideal step
+                dv = (vmax - vmin) / 3 * 24 << u.hourangle
+
+                # And round to nearest sensible value
+                dv = select_step_hour(dv).to_value(u.hourangle) / 24
+
+                # Determine tick locations
+                imin = np.ceil(vmin / dv)
+                imax = np.floor(vmax / dv)
+                return np.arange(imin, imax + 1, dtype=int) * dv
+
+        else:
+
+            return super().tick_values(vmin, vmax)
+
+    def __call__(self):
+        vmin, vmax = self.axis.get_view_interval()
+        return self.tick_values(vmin, vmax)
+
+
+class AstropyTimeFormatter(ScalarFormatter):
+
+    def __init__(self, converter, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._converter = converter
+        self.set_useOffset(False)
+        self.set_scientific(False)
+
+    def format_ticks(self, values):
+        if self._converter.format in YMDHMS_FORMATS:
+            if len(values) == 0:
+                return []
+            times = Time(values, format='mjd', scale=self._converter.scale)
+            formatted = getattr(times, self._converter.format)
+            if self._converter.simplify:
+                if all([x.endswith('00:00:00.000') for x in formatted]):
+                    split = ' ' if self._converter.format == 'iso' else 'T'
+                    formatted = [x.split(split)[0] for x in formatted]
+            return formatted
+        else:
+            return super().format_ticks(values)
+
+
+class MplTimeConverter(units.ConversionInterface):
+
+    def __init__(self, scale=None, format=None, simplify=None):
+
+        super().__init__()
+
+        self.format = format
+        self.scale = scale
+        self.simplify = simplify
+
+        if Time not in units.registry:
+            units.registry[Time] = self
+            self._remove = True
+        else:
+            self._remove = False
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, type, value, tb):
+        if self._remove:
+            del units.registry[Time]
+
+    def convert(self, value, unit, axis):
+        """
+        Convert a Time value to a scalar or array.
+        """
+        scaled = getattr(value, self.scale)
+        if self.format in YMDHMS_FORMATS:
+            return scaled.mjd
```

Comment:
Is this done to make the values numerical? But, if so, shouldn't it happen for all string formats? For best precision for relevant times, one might consider using `cxcsec` or so.

Reply:
I've now fixed compatibility for all string formats (and have tests for all of them).